### PR TITLE
Bump API version and increase unittest coverage

### PR DESCRIPTION
**Pull request by jwhonce**
* Change to libpod API 2.0.0
* Remove example code from coverage reports
* Add APIConnection ContextManager test

Signed-off-by: Jhon Honce <jhonce@redhat.com>

### CODE REVIEW

**Comment by jwhonce** [comment]
@mwhahaha PTAL

File: podman/api_connection.py [hunk 1]
```diff
@@ -90,14 +90,16 @@ def post(self, path, params=None, headers=None, encode=False):
             headers = {}
 
         if encode:
-            if "content-type" not in set(key.lower() for key in headers) and params:
+            if (
+                "content-type" not in set(key.lower() for key in headers)
+                and params
+            ):
```
Comment (mwhahaha) [inline podman/api_connection.py:96]: Is this an old formatting thing? I just ran black and I'm not getting this being updated. Also we switched to linelength 100 in #39 so this shouldn't be updated

Comment (jwhonce) [reply podman/api_connection.py]: @mwhahaha I updated my preferences for formatting.

**Review by mwhahaha — APPROVED** [review]
LGTM

**Pull request merged**